terraform/modules/network-validator/scripts/network_validation.sh

```diff
@@ -0,0 +1,1103 @@
+  validation_return_code=2
+fi
+
+# Check for Custom Resolver
+res=$(validate_dhcp_options ${WLS_SUBNET_OCID})
+if [[ $res -ne 0 ]]
+then
+  echo "WARNING: Missing OCI DNS Server [169.254.169.254] in the DNS Servers entries for the custom resolver"
+fi
+
+# Check if Internet Gateway exists in WLS subnet's VCN.
+res=$(validate_internet_gw_exist)
+if [[ $res -ne 0 ]]
+then
+  echo "WARNING: Missing internet gateway in the VCN of the WLS subnet [$WLS_SUBNET_OCID]"
+fi
+
+### Validation - Only when WLS Subnet OCID is provided ###
+
+if [[ -n ${WLS_SUBNET_OCID} && -z ${ADMIN_SRV_NSG_OCID} && -z ${MANAGED_SRV_NSG_OCID} ]]
+then
+  wls_subnet_cidr_block=$(oci network subnet get --subnet-id ${WLS_SUBNET_OCID} | jq -r '.data["cidr-block"]')
+
+  # Check if SSH port is open for access by WLS subnet CIDR
+  res=$(validate_subnet_port_access ${WLS_SUBNET_OCID} ${SSH_PORT} ${wls_subnet_cidr_block})
+
+  if [[ $res -ne 0 ]]
+  then
+    echo "ERROR: Port ${SSH_PORT} is not open for access by WLS Subnet CIDR [$wls_subnet_cidr_block] in WLS Subnet [$WLS_SUBNET_OCID]"
+    validation_return_code=2
+  fi
+
+  # Check if T3 Port is open for access by WLS subnet CIDR
+  res=$(validate_subnet_port_access ${WLS_SUBNET_OCID} ${T3_PORT} ${wls_subnet_cidr_block})
+  if [[ $res -ne 0 ]]
+  then
+    echo "ERROR: Port ${T3_PORT} is not open for access by WLS Subnet CIDR [$wls_subnet_cidr_block] in WLS Subnet [$WLS_SUBNET_OCID]"
+    validation_return_code=2
+  fi
+
+  # Check if Admin Console HTTP Port is open for access to ALL_IPS by WLS subnet CIDR
+  res=$(validate_subnet_port_access ${WLS_SUBNET_OCID} ${ADMIN_HTTP_PORT} ${ALL_IPS})
+  if [[ $res -eq 0 ]]
+  then
+    echo "WARNING: Exposing the WebLogic administrator port [${ADMIN_HTTP_PORT}] in the subnet [{$WLS_SUBNET_OCID}] to the internet [${ALL_IPS}] allows any user to access the WebLogic console, which is not a recommended practice. Ensure that only a specific CIDR range can access the WebLogic console."
+  fi
+
+  # Check if Admin Console HTTPS Port is open for access to ALL_IPS by WLS subnet CIDR
+  res=$(validate_subnet_port_access ${WLS_SUBNET_OCID} ${ADMIN_HTTPS_PORT} ${ALL_IPS})
+  if [[ $res -eq 0 ]]
+  then
+    echo "WARNING: Exposing the WebLogic administrator port [${ADMIN_HTTPS_PORT}] in the subnet [{$WLS_SUBNET_OCID}] to the internet [${ALL_IPS}] allows any user to access the WebLogic console, which is not a recommended practice. Ensure that only a specific CIDR range can access the WebLogic console."
+  fi
+fi
+
+### Validation - Only when WLS Subnet OCID, Admin Server NSG & Managed Server NSG are provided ###
+
+if [[ -n ${WLS_SUBNET_OCID} && -n ${ADMIN_SRV_NSG_OCID} && -n ${MANAGED_SRV_NSG_OCID} ]]
+then
+  wls_subnet_cidr_block=$(oci network subnet get --subnet-id ${WLS_SUBNET_OCID} | jq -r '.data["cidr-block"]')
+
+  # Check if SSH port is open for access by WLS subnet CIDR in Admin Server NSG
```

Comment:
this comment and the error message says admin server nsg but the managed one is used

Reply:
Fixed.